types/adnexo/ax-dist/uplink.js

```diff
@@ -0,0 +1,175 @@
+  }
+  return Bits.bitsToUnsigned(Bits.hexToBits(hexArray.join("")));
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const bits = Bits.hexToBits(payload);
+  const { port } = event.data;
+  const data = {};
+  let topic = "default";
+
+  if (port === 1) {
+    data.payloadType = "ECHO";
+    data.echo = payload;
+    topic = "lifecycle";
+  } else if (port === 2) {
+    data.payloadType = "RESTART";
+    topic = "lifecycle";
+  } else if (port === 3) {
+    data.payloadType = "NEW_CONFIGURATION_ACCEPTED";
+    topic = "lifecycle";
+  } else if (port === 4) {
+    let pointer = 0;
+
+    while (pointer < payload.length) {
+      const header = Bits.bitsToUnsigned(bits.substr(pointer * 4, 8));
+      pointer += 2;
+
+      switch (header) {
+        case 0x04:
+          data.measurementInterval = toLittleEndian(
+            payload.substr(pointer, 8),
+            false,
+          );
+          pointer += 8;
+          break;
+        case 0x05:
+          data.sendInterval = toLittleEndian(payload.substr(pointer, 2), false);
+          pointer += 2;
+          break;
+        case 0x06:
+          data.joinInterval = toLittleEndian(payload.substr(pointer, 8), false);
+          pointer += 8;
+          break;
+        case 0x07:
+          data.reedActive = !!toLittleEndian(payload.substr(pointer, 2), false);
+          pointer += 2;
+          break;
+        case 0x52:
+          data.thresholdDistance = toLittleEndian(
+            payload.substr(pointer, 4),
+            true,
+          );
+          pointer += 4;
+          break;
+
+        default:
+          pointer = payload.length;
+          break;
+      }
+    }
+
+    data.payloadType = "REPORT_CONFIGURATION";
+    topic = "lifecycle";
+  } else if (port === 5) {
+    const errorPort = Bits.bitsToUnsigned(bits.substr(0, 8));
+    const errorCode = Bits.bitsToUnsigned(bits.substr(0, 8));
+
+    if (errorPort === 2) {
+      data.errorOn = "RESTART";
+    } else if (errorPort === 3) {
+      data.errorOn = "PARAM_CHANGE";
+    } else if (errorPort === 4) {
+      data.errorOn = "PARAM_READ";
+    }
+
+    if (errorCode === 1) {
+      data.errorCode = "PARAM_UNKNOWN";
+    } else if (errorCode === 2) {
+      data.errorCode = "PARAM_READ_ONLY";
+    } else if (errorCode === 4) {
+      data.errorCode = "PARAM_NOT_READABLE";
+    } else if (errorCode === 8) {
+      data.errorCode = "COMMAND_STRUCTURE_WRONG";
+    }
+    topic = "error";
+  } else if (port === 100 || port === 101 || port === 102) {
+    const distance = toLittleEndian(payload.substr(0, 4), false) / 10;
+    if (distance !== 6553.5) {
+      data.distance = distance;
+    } else {
+      data.distance = null;
+    }
+
+    if (event.device !== undefined && data.distance !== undefined) {
+      if (event.device.customFields !== undefined) {
+        const scaleLength = Number(
+          event.device.customFields["Height of the tank in cm"],
```

Comment:
first make sure that the field with the given name exists before parsing it as number